### ISSUE TITLE
Various bugs I've run into within the hour

### ISSUE THREAD

**Issue by MakeSureToKnock** [TheUnderTaker11/GeneticsReborn#155]
I preface this by saying that I have the "keep genes on death" set to false in the config. Additionally, I tested all of this using commands, in survival.

Climb Walls:
It doesn't get removed on death or by command once added. Additionally, this gene persists through saves. However, restarting the game removes the gene from all saves.

Emerald Heart:
The player doesn't drop an emerald on death. This is most likely due to the "keep genes on death" being set to false.

Explosive Exit:
The player doesn't explode on death. Again, most likely due to the "keep genes on death" being set to false.

Item Magnet:
Upon dying, all dropped items will be sucked back into the dead player's inventory, causing them to be deleted upon respawn.

Infinity:
Shooting a bow will cause it to fire two arrows. Additionally, upon reloading a save, arrows fired with the infinity gene will have a blue potion particle effect attached to them as well as the second fired arrow disappearing.

Slimy:
Upon player death a slime does not spawn nor does the player get revived. Again, most likely due to the "keep genes on death" being set to false.

Teleporter:
Teleporter does not activate when pressing the appropriate key. Rebinding the key doesn't change this.

Efficiency:
Gene fails to work. This was tested in survival using the gene with a wooden pickaxe vs no gene and wooden pickaxe with efficiency 2. Coal was the medium.

No hunger:
Gene leaves your hunger at 2.5 shanks rather than 5. The wiki states half-way.

Scare Skeleton:
Does not scare skeletons.

Scare Spider:
Does not scare spiders.

Scare Zombie:
Does not scare zombies.

Thorns:
The damage of the gene is set to 0 innately in the config.

Claws:
The damage of the gene is set to 0 innately in the config.

I would suggest adding a command or item that lists the active genes that player has.

**Comment by clubpetey** [comment]
Thanks for the QA work, here are the results of your issues:

Climb Walls:
OnDeath event weren't firing correctly, plus this gene had some issues with syncing between the client and server.  It has been reported before, but is now fixed.

Emerald Heart:
OnDeath event weren't firing correctly, fixed in 1.24.

Explosive Exit:
OnDeath event weren't firing correctly, fixed in 1.24, also did you have at least 4 gun-powders in your inventory (in the same stack), otherwise this wont fire.

Item Magnet:
this one is tricky, I've added a check that the player is not dead before the magnets activate, but there is still SOME chance that playerTicks (and thus the magnet) may fire after you are dead, but before your "isDead" flag is set.  I'll have to play around with this, but it should be much better.

Infinity:
Moving to Issue #156 

Slimy:
OnDeath event weren't firing correctly, fixed in 1.24.

Teleporter:
Under investigation, this is tracked on issue #148 

Efficiency:
fixed, EFFICIENCY now doubles break speed, EFFICIENCY_4 multiplies by 5

No hunger:
fixed in 1.24, bad math on my part.

Scare Skeleton:
fixed in 1.24

Scare Spider:
fixed in 1.24

Scare Zombie:
fixed in 1.24

Thorns:
default was changed, but as you have a config, you will need to reset it to 6.0 or delete your config

Claws:
default was changed, but as you have a config, you will need to reset it to 1.0 or delete your config

Suggest adding a command or item that lists the active genes that player has...
This already exists, draw your blood with a syringe and the tool tip will tell you the genes in your blood.
